outlines/fsm/fsm.py

```diff
@@ -189,17 +191,28 @@ def copy(self) -> "RegexFSM":
 class CFGFSM(FSM):
     """FSM to generate text that is in the language of a context-free grammar."""
 
-    def __init__(self, cfg_string: str, tokenizer: "Tokenizer"):
-        self.cfg_string = cfg_string
+    def __init__(self, cfg: Union[str, Path], tokenizer: "Tokenizer"):
+        self.cfg = cfg
         self.tokenizer = tokenizer
 
+        if isinstance(cfg, Path):
```

Comment:
Could we load the grammars as strings instead of adding code that is not directly related to building the FSM here?

Reply:
Great idea, much cleaner. Done.